File: scala/scala3/CSharp/Scala3ParserBase.cs
#nullable enable
using Antlr4.Runtime;
using System.IO;

public abstract class Scala3ParserBase : Parser
{
    protected Scala3ParserBase(ITokenStream input) : base(input) { }
    protected Scala3ParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput) { }

    /// <summary>
    /// Returns true when the --3.0-migration command-line flag is set,
    /// enabling Scala 2-compatible syntax (._  wildcard imports, [_] type wildcards).
    /// </summary>
    protected bool migration30() => Scala3LexerBase.Migration30;
}

File: scala/scala3/Java/Scala3ParserBase.java
import org.antlr.v4.runtime.Parser;
import org.antlr.v4.runtime.TokenStream;

public abstract class Scala3ParserBase extends Parser {

    protected Scala3ParserBase(TokenStream input) {
        super(input);
    }

    /**
     * Returns true when the --3.0-migration command-line flag is set,
     * enabling Scala 2-compatible syntax (._ wildcard imports, [_] type wildcards).
     */
    protected boolean migration30() {
        return Scala3LexerBase.migration30;
    }
}
